src/views/Login.vue

```diff
@@ -1,39 +1,101 @@
+          </b-form-group>
+
+          <b-form-group label="Password:" label-for="password">
+            <b-form-input id="password" v-model="form.password" type="password" required :disabled="loading"></b-form-input>
+          </b-form-group>
+
+          <b-alert v-if="error" variant="danger" show>
+            {{ error }}
+          </b-alert>
+
+          <b-button type="submit" variant="primary" :disabled="loading" class="w-100">
+            <b-spinner v-if="loading" small></b-spinner>
+            {{ loading ? 'Logging in...' : 'Login' }}
+          </b-button>
+        </b-form>
+      </b-card>
+      <div class="mt-3 text-center">
+        <!-- TODO: Translate this -->
+        <p>Forgot your password?<router-link :to="{ name: 'passwordReset' }"> Reset it</router-link>.</p>
+        <!-- TODO: Translate this -->
+        <p>Not a member?<router-link :to="{ name: 'register' }"> Register</router-link>.</p>
+        <p>
+          Use of any LCO facilities and/or data implies acceptance of the
+          <a href="https://lco.global/observatory/termsofservice/"> Terms of Service</a>.
+        </p>
+      </div>
     </b-col>
   </b-row>
 </template>
 <script>
-import PassthroughForm from '@/components/PassthroughForm.vue';
-
 export default {
   name: 'Login',
-  components: {
-    PassthroughForm
+  props: {
+    redirectPath: {
+      type: String,
+      default: null
+    }
   },
-  computed: {
-    redirectPath: function() {
-      return this.$route.query.next || '/';
+  data() {
+    return {
+      form: {
+        username: '',
+        password: ''
+      },
+      loading: false,
+      error: null
     }
   },
   methods: {
-    onFormUpdated: function(form) {
-      // Login form can also redirect to change password form.
-      // If the input button says "Change password" redirect to the
-      // correct Vue route. Otherwise this form will post to the wrong endpoint.
-      if (form.find("input[value='Change password']").length == 1)
-          this.$router.push({name: "passwordChange"})
+    async handleSubmit() {
+      this.loading = true;
+      this.error = null;
+
+      try {
+        await this.loginUser();
+        await this.loadProfile();
+        this.redirectAfterLogin();
+      } catch (err) {
+        this.error = err.message;
+      } finally {
+        this.loading = false;
+      }
+    },
+    async loginUser() {
+      const response = await fetch(`${this.$store.state.urls.observationPortalApi}/accounts/api-login/`, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json'
+        },
+        credentials: 'include', // set cookies
+        body: JSON.stringify(this.form)
+      });
+      if (!response.ok) {
+        const data = await response.json();
+        throw new Error(data.message || 'Login failed');
+      }
+    },
+    async loadProfile() {
+      try {
+        await this.$store.dispatch('getProfileData');
```

Comment:
The store getProfileData function is not `async`, so I don't think you can `await` it. It does return a `Promise` though so you can call `.then(...)` on the returned value.

Reply:
hmm okay